File: C#(MAIN WORK)/src/OrderDomain/Entities/Product.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace OrderDomain.Entities
{
    public class Product
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        public string ImageUrl { get; set; }
        public string Name { get; set; }
        public double Price { get; set; }
        
    }
}

File: GO(SECONDARY WORK)/src/Services/OrderService/types/Product.go
package types

type Product struct {
	Id       string  `bson:"_id" json:"_id" validate:"isdefault"`
	ImageUrl string  `bson:"imageUrl" json:"imageUrl" validate:"required"`
	Name     string  `bson:"name" json:"name" validate:"required"`
	Price    float32 `bson:"price" json:"price" validate:"required"`
}
